### ISSUE TITLE
StaticArray equality

### ISSUE THREAD

**Issue by stugol** [manastech/crystal#1205]
Shouldn't the following code print ```true```?
```
a = StaticArray(Int32, 3).new { |n| n }
b = StaticArray(Int32, 3).new { |n| n }
puts a == b
```
(http://play.crystal-lang.org/#/r/bm8)

**Comment by waj** [comment]
StaticArray#== is not implemented (yet), and the default implementation for object references returns true if they are the exact same instance.

**Comment by stugol** [comment]
I've monkeypatched an == operator into ```StaticArray(T,_)``` for my own use; but I'm not sure how efficient it might be. It uses ```zip``` to iterate through each value.

**Comment by asterite** [comment]
This is implemented now thanks to #1250
